### PR TITLE
codegen_scalar: support tuples with more than one element

### PR DESCRIPTION
**Pull request by tautschnig**
### Description of changes: 

With toolchain versions from 2023-02-05 onwards we have multi-element tuples in a Scalar (seen for the `(DropMe, Yielded(()))` tuple of tests/kani/Generator/rustc-generator-tests/smoke-resume-args.rs).

### Resolved issues:

n/a

### Related RFC:

n/a

### Call-outs:

n/a

### Testing:

* How is this change tested? The actual use of multi-element tuples might only be seen with future toolchains (cf. #2347).

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- n/a Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 1]
```diff
@@ -394,15 +394,14 @@ impl<'tcx> GotocCtx<'tcx> {
...
-                // here we have tuples of at most one length
-                if substs.len() == 1 {
-                    let overall_t = self.codegen_ty(ty);
-                    let t = substs[0];
-                    self.codegen_single_variant_single_field(s, span, overall_t, t)
-                } else {
-                    unreachable!()
-                }
+            (Scalar::Int(int), ty::Tuple(_)) => {
+                let int_u128 = int.try_to_uint(int.size()).ok().unwrap();
```
Comment (zhassan-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:398]: Is this guaranteed to fit in 128 bits? If so, can we add a link to the upstream PR/code that makes this assumption?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 2]
```diff
@@ -394,15 +394,14 @@ impl<'tcx> GotocCtx<'tcx> {
...
-            (s, ty::Tuple(substs)) => {
-                // here we have tuples of at most one length
-                if substs.len() == 1 {
-                    let overall_t = self.codegen_ty(ty);
-                    let t = substs[0];
-                    self.codegen_single_variant_single_field(s, span, overall_t, t)
-                } else {
-                    unreachable!()
-                }
+            (Scalar::Int(int), ty::Tuple(_)) => {
```
Comment (zhassan-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:397]: Can we update the doc comment on this function and mention that multi-element tuples are included (in `4.`)?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 1 above)
Comment (tautschnig) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:401]: I've added a comment to the code.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 2 above)
Comment (tautschnig) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:397]: Done.

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**